src/commands/protocols/common/submit-commit-command.js

```diff
@@ -109,107 +118,40 @@ class SubmitCommitCommand extends Command {
-                blockchain,
-                error.message,
-                ERROR_TYPE.COMMIT_PROOF.SUBMIT_COMMIT_SEND_TX_ERROR,
-            );
-            let newGasPrice;
-            if (
-                error.message.includes(`timeout exceeded`) ||
-                error.message.includes(`Pool(TooLowPriority`)
-            ) {
-                newGasPrice = Math.ceil(txGasPrice * COMMAND_TX_GAS_INCREASE_FACTORS.SUBMIT_COMMIT);
-            } else {
-                newGasPrice = null;
-            }
-
-            Object.assign(command, {
-                data: { ...command.data, gasPrice: newGasPrice },
-                message: error.message,
-            });
-
-            return Command.retry();
-        }
-
-        let msgBase;
-        if (txSuccess) {
-            this.operationIdService.emitChangeEvent(
-                OPERATION_ID_STATUS.COMMIT_PROOF.SUBMIT_COMMIT_SEND_TX_END,
-                sendSubmitCommitTransactionOperationId,
-                blockchain,
-                agreementId,
-                epoch,
-                operationId,
-            );
-            msgBase = 'Successfully executed';
-
-            this.operationIdService.emitChangeEvent(
-                OPERATION_ID_STATUS.COMMIT_PROOF.SUBMIT_COMMIT_END,
-                operationId,
-                blockchain,
-                agreementId,
-                epoch,
-            );
-        } else {
-            msgBase = 'Node has already submitted commit. Finishing';
-            this.operationIdService.emitChangeEvent(
-                OPERATION_ID_STATUS.FAILED,
-                sendSubmitCommitTransactionOperationId,
-                blockchain,
-                msgBase,
-                ERROR_TYPE.COMMIT_PROOF.SUBMIT_COMMIT_SEND_TX_ERROR,
-            );
-        }
-
-        this.logger.trace(
-            `${msgBase} ${command.name} for the Service Agreement with the ID: ${agreementId}, ` +
-                `Blockchain: ${blockchain}, Contract: ${contract}, Token ID: ${tokenId}, ` +
-                `Keyword: ${keyword}, Hash function ID: ${hashFunctionId}, Epoch: ${epoch}, ` +
-                `State Index: ${stateIndex}, Operation ID: ${operationId}, ` +
-                `Closest Node: ${closestNode}, Left neighborhood edge: ${leftNeighborhoodEdge}, ` +
-                `Right neighborhood edge: ${rightNeighborhoodEdge}, ` +
-                `Retry number: ${COMMAND_RETRIES.SUBMIT_COMMIT - command.retries + 1}`,
+                closestNode,
+                leftNeighborhoodEdge,
+                rightNeighborhoodEdge,
+                contract,
+                tokenId,
+                keyword,
+                hashFunctionId,
+                stateIndex,
+                txGasPrice,
+            },
+            command,
         );
+    }
 
-        return Command.empty();
+    async insufficientFundsErrorReceived(commandData) {
+        await this.repositoryModuleManager.updateServiceAgreementLastCommitEpoch(
```

Comment:
it might make sense to use the same logic on last failed retry, if the failure is not due to commit window / already committed errors.